components/builder-db/src/models/projects.rs

```diff
@@ -66,15 +66,20 @@ pub struct UpdateProject<'a> {
 }
 
 impl Project {
-    pub fn get(name: &str, conn: &PgConnection) -> QueryResult<Project> {
+    pub fn get(name: &str, target: &str, conn: &PgConnection) -> QueryResult<Project> {
         Counter::DBCall.increment();
         origin_projects::table.filter(origin_projects::name.eq(name))
+                              .filter(origin_projects::target.eq(target))
                               .get_result(conn)
     }
 
-    pub fn delete(name: &str, conn: &PgConnection) -> QueryResult<usize> {
+    pub fn delete(name: &str, target: &str, conn: &PgConnection) -> QueryResult<usize> {
         Counter::DBCall.increment();
-        diesel::delete(origin_projects::table.filter(origin_projects::name.eq(name))).execute(conn)
+        if target.to_string().eq("*") {
```

Comment:
I'm not sure about the "*".  I believe we don't have a use case (eg UI) planned for this currently.  If we do really want this, I think we'd be better off with something more structured or even a different API route just so that there's no inadvertent use of the delete all.